src/sentry/static/sentry/app/views/organizationEventsV2/transactionView/dividerHandlerManager.tsx

```diff
@@ -0,0 +1,221 @@
+    if (this.isDragging) {
+      return;
+    }
+
+    selectRefs(this.dividerLineRefs, dividerDOM => {
+      if (nextHover) {
+        dividerDOM.classList.add('hovering');
+        return;
+      }
+
+      dividerDOM.classList.remove('hovering');
+    });
+  };
+
+  onDragStart = (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+    if (this.isDragging || event.type !== 'mousedown' || !this.hasInteractiveLayer()) {
+      return;
+    }
+
+    event.stopPropagation();
+
+    // prevent the user from selecting things outside the minimap when dragging
+    // the mouse cursor outside the minimap
+
+    this.previousUserSelect = setBodyUserSelect({
+      userSelect: 'none',
+      MozUserSelect: 'none',
+      msUserSelect: 'none',
+    });
+
+    // attach event listeners so that the mouse cursor does not select text during a drag
+    window.addEventListener('mousemove', this.onDragMove);
+    window.addEventListener('mouseup', this.onDragEnd);
+
+    // indicate drag has begun
+
+    this.isDragging = true;
+
+    selectRefs(this.dividerLineRefs, (dividerDOM: HTMLDivElement) => {
+      dividerDOM.style.backgroundColor = 'rgba(73,80,87,0.75)';
+    });
+
+    selectRefs(this.ghostDividerLineRefs, (dividerDOM: HTMLDivElement) => {
+      dividerDOM.style.display = 'block';
+    });
+  };
+
+  onDragMove = (event: MouseEvent) => {
+    if (!this.isDragging || event.type !== 'mousemove' || !this.hasInteractiveLayer()) {
+      return;
+    }
+
+    const rect = rectOfContent(this.props.interactiveLayerRef.current!);
+
+    // mouse x-coordinate relative to the interactive layer's left side
+    const rawMouseX = (event.pageX - rect.x) / rect.width;
+
+    const min = 0;
+    const max = 1;
+
+    // clamp rawMouseX to be within [0, 1]
+    this.dividerHandlePosition = clamp(rawMouseX, min, max);
+
+    const dividerHandlePositionString = toPercent(this.dividerHandlePosition);
+
+    selectRefs(this.dividerLineRefs, (dividerDOM: HTMLDivElement) => {
+      dividerDOM.style.left = dividerHandlePositionString;
+    });
+  };
+
+  onDragEnd = (event: MouseEvent) => {
+    if (!this.isDragging || event.type !== 'mouseup' || !this.hasInteractiveLayer()) {
+      return;
+    }
+
+    // remove listeners that were attached in onDragStart
+
+    this.cleanUpListeners();
+
+    // restore body styles
+
+    if (this.previousUserSelect) {
+      setBodyUserSelect(this.previousUserSelect);
+      this.previousUserSelect = null;
+    }
+
+    // indicate drag has ended
+
+    this.isDragging = false;
+
+    selectRefs(this.dividerLineRefs, (dividerDOM: HTMLDivElement) => {
+      dividerDOM.style.backgroundColor = null;
+    });
+
+    selectRefs(this.ghostDividerLineRefs, (dividerDOM: HTMLDivElement) => {
+      dividerDOM.style.display = 'none';
+    });
+
+    this.setState({
+      // commit dividerHandlePosition to be dividerPosition
+      dividerPosition: this.dividerHandlePosition,
+    });
+  };
+
+  cleanUpListeners = () => {
```

Comment:
Should we also cleanup on `componentWillUnmount`?

Reply:
Whoops. I thought I did this already.